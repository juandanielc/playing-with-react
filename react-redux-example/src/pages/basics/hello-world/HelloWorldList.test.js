import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HelloWorldList, { HelloWorld } from './index';
import AddGreeter from './AddGreeter';

describe(HelloWorldList, () => {
  const component = shallow(
    <HelloWorldList />
  );


  	// 1. Writing A Generic Snapshot Test
  	
	it('renders and matches our snapshot', () => {
	  const component = renderer.create(
	    <HelloWorldList />
	  );
	  const tree = component.toJSON();
	  expect(tree).toMatchSnapshot();
	});

	// 2. Writing Content Tests

	it('contains an AddGreeter subcomponent', () => {
	  expect(component.find(AddGreeter)).toHaveLength(1);
	});

	it('contains the same number of HelloWorld components as greetings', () => {
	  const helloWorlds = component.find(HelloWorld).length;
	  const greetings   = component.state('greetings').length;
	  // console.log('helloWorlds', helloWorlds); It is 3 for the initial names 
	  expect(helloWorlds).toEqual(greetings);
	});

	// 3. Checking Interactions
	
	it('adds another greeting when the add greeting function is called', () => {
	  const before = component.find(HelloWorld).length;
	  component.instance().addGreeting('Sample');
	  const after = component.find(HelloWorld).length;
	  expect(after).toBeGreaterThan(before);
	});

	it('removes a greeting from the list when the remove greeting function is called', () => {
	  const before   = component.find(HelloWorld).length;
	  const removeMe = component.state('greetings')[0];
	  component.instance().removeGreeting(removeMe);
	  const after = component.find(HelloWorld).length;
	  expect(after).toBeLessThan(before);
	});
});