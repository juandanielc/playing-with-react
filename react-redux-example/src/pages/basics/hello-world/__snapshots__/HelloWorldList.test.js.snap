// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorldList extends _react.Component {

	constructor(props) {
		super(props);
		this.state = { greetings: ['Jim', 'Sally', 'juan'] };
		this.addGreeting = this.addGreeting.bind(this);
		this.removeGreeting = this.removeGreeting.bind(this);
	}

	renderGreetings() {
		return this.state.greetings.map(name =>
		_react2.default.createElement(HelloWorld, { key: name, name: name, removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 46 }, __self: this }));

	}

	addGreeting(newName) {
		this.setState({ greetings: [...this.state.greetings, newName] });
	}

	removeGreeting(removeName) {
		const filteredGreetings = this.state.greetings.filter(name => {
			return name !== removeName;
		});
		this.setState({ greetings: filteredGreetings });
	}

	render() {
		return (
			_react2.default.createElement('div', { className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 63 }, __self: this },
				_react2.default.createElement('p', { __source: { fileName: _jsxFileName, lineNumber: 64 }, __self: this }, _react2.default.createElement('strong', { __source: { fileName: _jsxFileName, lineNumber: 64 }, __self: this }, 'Check the tests into the folder src/pages/basics/hello-world')),
				_react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 65 }, __self: this }),
				this.renderGreetings()));


	}} renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <p>
    <strong>
      Check the tests into the folder src/pages/basics/hello-world
    </strong>
  </p>
  <div
    className="AddGreeter"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
      
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Jim
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Greeting
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Sally
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Greeting
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    juan
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove Greeting
    </button>
  </div>
</div>
`;
